File: org.inqle.ui.google.widgets/src/org/inqle/ui/google/jsapi/ColumnChart.js
/*******************************************************************************
 * Copyright (c) 2002-2008 Innoopract Informationssysteme GmbH.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     Innoopract Informationssysteme GmbH - initial API and implementation
 ******************************************************************************/

qx.Class.define( "org.inqle.ui.google.jsapi.ColumnChart", {
    extend: qx.ui.layout.CanvasLayout,
    
    construct: function( id ) {
        this.base( arguments );
        this.setHtmlAttribute( "id", id );
        this._id = id;
        this._chart = null;
        
        this.addEventListener( "changeHeight", this._doResize, this );
        this.addEventListener( "changeWidth", this._doResize, this );
        
    },
    
    properties : {
        widgetData : {
            init : "",
            apply : "load"
        }
    },
    
    members : {
        _doActivate : function() {
            var shell = null;
            var parent = this.getParent();
            while( shell == null && parent != null ) {
                if( parent.classname == "org.eclipse.swt.widgets.Shell" ) {
                    shell = parent;
                }
                parent = parent.getParent();
            }
            if( shell != null ) {
                shell.setActiveChild( this );
            }
        },
        
        load : function() {
	    	qx.ui.core.Widget.flushGlobalQueues();
	    	var theData = eval('(' + this.getWidgetData() + ')');
	        if( this._chart == null ) {
	            this._chart = new google.visualization.ColumnChart(document.getElementById(this._id));
	//                	this._chart = new google.visualization.ColumnChart(this.getElement());
	//                    this._chart.addControl( new GSmallMapControl() );
	//                    this._chart.addControl( new GMapTypeControl() );
	//                    GEvent.bind( this._chart, "click", this, this._doActivate );
	//                    GEvent.bind( this._chart, "moveend", this, this._onMapMove );
	            
	        }
	        var dataTable  = new google.visualization.DataTable(theData);
	        
	        var chart = this._chart;
	        var _inqlegv_options = {};
	        chart.draw(dataTable, _inqlegv_options);
        },
        
        _doResize : function() {
//            qx.ui.core.Widget.flushGlobalQueues();
//            if( this._chart != null ) {
//                this._chart.checkResize();
//            }
        }
    }
    
} );

File: org.inqle.ui.google.widgets/src/org/inqle/ui/google/jsapi/ColumnChart.java
/*******************************************************************************
 * Copyright (c) 2002-2008 Innoopract Informationssysteme GmbH.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     Innoopract Informationssysteme GmbH - initial API and implementation
 ******************************************************************************/
package org.inqle.ui.google.jsapi;

import org.eclipse.swt.widgets.Composite;

/**
 * Renders a Google Visualization Column Chart.
 * 
 * Note that this widget is rendered upon calling the setWidgetData method.  
 * So if you wish to set options like width, height, colors, etc., you must do this 
 * using method setWidgetOptions, before calling setWidgetData.
 * 
 * Usage:
 * JSONGoogleDataTable dataTable = new JSONGoogleDataTable();
    dataTable.addColumn("theyear", "Date", "string", null);
    dataTable.addColumn("CO2", "CO2", "number", null);
    dataTable.addColumn("Temperature", "Temperature", "number", null);
    dataTable.addRow(new Object[] {"1970", 300, 22});
    dataTable.addRow(new Object[] {"2009", 400, 24});
    String serializedData = dataTable.toString();
    
    l = new Label(composite, SWT.NONE);
    l.setText("Column Chart");
    ColumnChart chart = new ColumnChart( composite, SWT.NONE );
    chart.setWidgetOptions("{width: 300, height: 300}");
    chart.setWidgetData(serializedData);
    gridData = new GridData(300, 300);
    chart.setLayoutData(gridData);
    
 * @See http://code.google.com/apis/visualization/documentation/gallery/columnchart.html
 * @author David Donohue
 * 2009/4/8
 */
public class ColumnChart extends VisualizationWidget {

  public ColumnChart( final Composite parent, final int style ) {
    super( parent, style );
  }

}
